File: Data Structures & Algorithms/maximum-subarray/submission-0.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sol = Integer.MIN_VALUE, curr = 0;
        for (int i = 0; i < nums.length; i++) {
            curr += nums[i];
            sol = Math.max(sol, curr);
            if (curr < 0) curr = 0;
        }
        return sol;
    }
}

File: Data Structures & Algorithms/subsets/submission-0.go
func subsets(nums []int) [][]int {
    n, sol := len(nums), [][]int{}

    var bt func (start int, curr []int)
    bt = func (start int, curr []int) {
        sol = append(sol, append([]int{}, curr...))
        if start == n {
            return
        }
        for i := start; i < n; i++ {
            curr = append(curr, nums[i])
            bt(i+1, curr)
            curr = curr[:len(curr)-1]
        }
    }

    bt(0, []int{})
    return sol
}
